### PR TITLE
Add IVAs to access request management

### PR DESCRIPTION
**Pull request by ac-jorellanaf**
- Closes GSI-644

### CODE REVIEW

File: src/mocks/data.js [hunk 1]
```diff
@@ -8,6 +8,19 @@ export const user = {
   email: "j.jdoe@home.org",
 };
 
+export const IVAs = [{
+  id: "TEST00014000001",
+  type: "Phone",
+  value: "+441234567890004",
+  status: "Verified"
```
Comment (Cito) [inline src/mocks/data.js:15]: As a side note: We need to check that the status enums are translated properly from the backend to the frontend, and that we use enum types consistently throughout the application. Probably after the backend work has been finished.

(The difficulties arise from the different naming conventions in Python and JavaScript and the different kinds of enum types or string literal types on both sides that could be used.)

**Review by Cito — CHANGES_REQUESTED** [review]
Looks good overall, but some details need more attention - see my suggestions.

File: src/mocks/data.js (hunk 1 above)
Comment (ac-jorellanaf) [reply src/mocks/data.js:15]: I tackle this in the PR for 642. So we can resolve it here hopefully?

File: src/components/accessRequests/accessRequestsList/accessRequestModal.tsx [hunk 2]
```diff
@@ -148,38 +152,125 @@ const AccessRequestModal = (props: AccessRequestModalProps) => {
...
+                {props.userIVAs.length > 0 ? (
+                  <>
+                    {props.userIVAs.map((x: IVA) => (
+                      <div key={x.id}>
+                        <Row className="mb-1">
+                          <Col xs={7}>
+                            <input
+                              type="radio"
+                              className="me-1"
+                              id={x.id}
```
[the remark this answers is not in the document]
Comment (ac-jorellanaf) [reply src/components/accessRequests/accessRequestsList/accessRequestModal.tsx]: > Should we emphasize the selected option (using bold text, different color, or border?)

Done 👍

File: src/components/accessRequests/accessRequestsList/accessRequestModal.tsx [hunk 3]
```diff
@@ -148,38 +152,125 @@ const AccessRequestModal = (props: AccessRequestModalProps) => {
...
+                              value={x.id}
+                              onClick={() => setSelectedIVA(x.id)}
+                            />
+                            <label htmlFor={x.id}>
+                              {x.type}: {x.value}
+                            </label>
+                          </Col>
+                          <Col>
+                            <label
+                              htmlFor={x.id}
```
[the remark this answers is not in the document]
Comment (ac-jorellanaf) [reply src/components/accessRequests/accessRequestsList/accessRequestModal.tsx:196]: Done 👍

File: src/components/accessRequests/accessRequestsList/accessRequestModal.tsx [hunk 4]
```diff
@@ -148,38 +152,125 @@ const AccessRequestModal = (props: AccessRequestModalProps) => {
...
+                              }
+                            >
+                              {x.status}
+                            </label>
+                          </Col>
+                        </Row>
+                      </div>
+                    ))}
+                    <p className="mt-3 mb-0">
+                      Please select the verification address that should be used
```
[the remark this answers is not in the document]
Comment (ac-jorellanaf) [reply src/components/accessRequests/accessRequestsList/accessRequestModal.tsx]: Done 👍

File: src/components/accessRequests/accessRequestsList/accessRequestModal.tsx [hunk 5]
```diff
@@ -148,38 +152,125 @@ const AccessRequestModal = (props: AccessRequestModalProps) => {
...
+                      to secure the access request
+                    </p>
+                  </>
+                ) : (
+                  "No verification addresses have been entered by the user so far"
```
[the remark this answers is not in the document]
Comment (ac-jorellanaf) [reply src/components/accessRequests/accessRequestsList/accessRequestModal.tsx]: Done 👍

File: src/components/accessRequests/accessRequestsList/accessRequestModal.tsx [hunk 6]
```diff
@@ -148,38 +152,125 @@ const AccessRequestModal = (props: AccessRequestModalProps) => {
...
+          ) : (
+            <></>
+          )}
+          {props.accessRequest?.status === "allowed" ? (
+            <Row className={ROW_CLASSES}>
+              <Col>
+                Verification address used:
+                <br />
+                {props.userIVAs.find(
+                  (x: IVA) => x.id === props.accessRequest?.iva
```
[the remark this answers is not in the document]
Comment (ac-jorellanaf) [reply src/components/accessRequests/accessRequestsList/accessRequestModal.tsx]: Done 👍

File: src/components/accessRequests/accessRequestsList/accessRequestsList.tsx [hunk 7]
```diff
@@ -107,11 +113,41 @@ const AccessRequestsList = (props: AccessRequestListProps) => {
...
+    url = new URL(`users/${props.user.id}/ivas`, WPS_URL);
+    let method: string = "GET",
+      ok: number = 200;
+    const response = await fetchJson(url, method).catch(() => null);
+    if (response && response.status === ok) {
+      try {
+        const IVAs = await response.json();
+        setUserIVAs(IVAs);
+      } catch {
+        showMessage({
```
[the remark this answers is not in the document]
Comment (ac-jorellanaf) [reply src/components/accessRequests/accessRequestsList/accessRequestsList.tsx:131]: Done 👍

File: src/components/accessRequests/accessRequestsList/accessRequestModal.tsx [hunk 8]
```diff
@@ -148,38 +152,125 @@ const AccessRequestModal = (props: AccessRequestModalProps) => {
                 : "Access request is pending"}
             </Col>
           </Row>
+          {props.accessRequest?.status === "pending" ? (
+            <Row className={ROW_CLASSES}>
+              <Col className="mt-3">
+                {props.userIVAs.length > 0 ? (
+                  <>
```
[the remark this answers is not in the document]
Comment (ac-jorellanaf) [reply src/components/accessRequests/accessRequestsList/accessRequestModal.tsx:174]: Forgot to comment, but this is also done 👍

File: src/components/accessRequests/accessRequestsList/accessRequestModal.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Cito) [reply src/components/accessRequests/accessRequestsList/accessRequestModal.tsx]: Still, in this case the user has to click on the IVA to select it, although there is only one. Users might find this unintuitive. Generally, a single radio button is always awkward.Maybe pre-select the single radio button in this case, or make it a checkbox instead? Any better ideas?

File: src/components/accessRequests/accessRequestsList/accessRequestModal.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (Cito) [reply src/components/accessRequests/accessRequestsList/accessRequestModal.tsx:174]: This is still unresolved.

**Review by Cito — CHANGES_REQUESTED** [review]

File: src/components/accessRequests/accessRequestsList/accessRequestsList.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (ac-jorellanaf) [reply src/components/accessRequests/accessRequestsList/accessRequestsList.tsx:131]: Not for me 🤔

File: src/components/accessRequests/accessRequestsList/accessRequestModal.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (ac-jorellanaf) [reply src/components/accessRequests/accessRequestsList/accessRequestModal.tsx]: I improved behaviour here

File: src/components/accessRequests/accessRequestsList/accessRequestModal.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (ac-jorellanaf) [reply src/components/accessRequests/accessRequestsList/accessRequestModal.tsx:176]: It seems I actually had not done it. It should be fine now

[the remark this answers is not in the document]
Comment (Cito) [reply src/components/accessRequests/accessRequestsList/accessRequestModal.tsx:176]: Thanks. Looks good now.

File: src/components/accessRequests/accessRequestsList/accessRequestsList.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (Cito) [reply src/components/accessRequests/accessRequestsList/accessRequestsList.tsx]: Pushed 6c00d882 to show what I meant.

**Review by Cito — APPROVED** [review]
Pushed 3 more commits. From my side, this is ready to be merged.

**Pull request merged**